src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowHistorySpec.groovy

```diff
@@ -0,0 +1,125 @@
+
+    @Autowired
+    @Shared
+    HaFlowHelper haFlowHelper
+
+    @Tidy
+    def "History records are created for the create"() {
+        given: "Ha flow"
+        def swT = topologyHelper.getAllNotNeighbouringSwitchTriplets().shuffled().first()
+        def haFlow = haFlowHelper.addHaFlow(haFlowHelper.randomHaFlow(swT))
+
+        when: "Ha Flow history contains create event"
+        //implement method
+        haFlowHelper.getHistory(haFlow.haFlowId).getCreateEntries().size() == 1
+
+        then: "Ha flow possible to be validated"
+        haFlowHelper.validate(haFlow.haFlowId)
+
+        cleanup:
+        haFlowHelper.deleteHaFlow(haFlow.haFlowId)
+    }
+    @Ignore("https://github.com/telstra/open-kilda/issues/5320")
+    @Tidy
+    def "User can change a ha-flow and get its history event - #description"() {
+        given: "Existing ha-flow"
+        def swT = topologyHelper.switchTriplets[0]
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT)
+        def haFlow = haFlowHelper.addHaFlow(haFlowRequest)
+
+        when: "#description the flow"
+        change(haFlow)
+
+        then: "Ha flow update event appears"
+        expectedAssertion(haFlow)
+
+        cleanup:
+        haFlowHelper.deleteHaFlow(haFlow.haFlowId)
+
+        where:
+        description | change | expectedAssertion
+
+        "update"    |
+                { HaFlow flow ->
+                    def allowedSharedPorts = topology.getAllowedPortsForSwitch(topology.find(
+                            flow.sharedEndpoint.switchId)) - flow.sharedEndpoint.portNumber
+                    flow.sharedEndpoint.portNumber = allowedSharedPorts[0]
+                    def updatedHaFlowPayload = haFlowHelper.convertToUpdate(flow)
+                    haFlowHelper.updateHaFlow(flow.haFlowId, updatedHaFlowPayload)
+
+                }            | { HaFlow flow -> haFlowHelper.getHistory(flow.haFlowId).getUpdateEntries().size() == 1}
+        "delete"    |
+                { HaFlow flow ->
+                    haFlowHelper.deleteHaFlow(flow.haFlowId)
+
+                }            | { HaFlow flow -> haFlowHelper.getHistory(flow.haFlowId).getDeleteEntries().size() == 1}
+
+        "reroute"    |
+                { HaFlow flow ->
+                    haFlowHelper.rerouteHaFlow(flow.haFlowId)
+
+                }            | { HaFlow flow -> haFlowHelper.getHistory(flow.haFlowId).getRerouteEntries().size() == 1}
+    }
+
+
+    def "History records can be get with timestamp filters"() {
+        given: "Ha flow"
+        def timestampBeforeCreate = System.currentTimeSeconds()
+        def swT = topologyHelper.getAllNotNeighbouringSwitchTriplets().shuffled().first()
+        def haFlow = haFlowHelper.addHaFlow(haFlowHelper.randomHaFlow(swT))
+
+        when: "Delete HA flow"
+        def timestampBeforeDelete = System.currentTimeSeconds()
+        haFlowHelper.deleteHaFlow(haFlow.haFlowId)
+        Wrappers.wait(WAIT_OFFSET) { assert !northboundV2.getHaFlow(haFlow.haFlowId) }
```

Comment:
Don't call Northbound from test specs. Add `deleteHaFlow()` method with verification to HaFlowHelper.

Reply:
removed cause the previous method already checked it